core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -282,15 +214,10 @@ public void deleteStore(String keyStoreName) throws SecurityException {
         if (KeyStoreUtil.isTrustStore(keyStoreName)) {
             throw new SecurityException("Not allowed to delete the trust store : " + keyStoreName);
         }
-        String path = RegistryResources.SecurityManagement.KEY_STORES + "/" + keyStoreName;
         try {
-            org.wso2.carbon.registry.api.Association[] assocs = registry.getAllAssociations(path);
-            if (assocs.length > 0) {
-                throw new SecurityException("Key store : " + keyStoreName + " is already in use and can't be deleted");
-            }
-            registry.delete(path);
+            keyStoreRegistryPersistenceManager.deleteKeyStore(keyStoreName);
             deleteKeyStoreFromCache(keyStoreName);
-        } catch (org.wso2.carbon.registry.api.RegistryException e) {
+        } catch (SecurityException e) {
```

Comment:
Whta are we catching and throwing the same exception?

Reply:
Fixed in the latest changes